xbmc/cores/paplayer/AudioDecoder.cpp

```diff
@@ -58,7 +58,10 @@ void CAudioDecoder::Destroy()
   m_pcmBuffer.Destroy();
 
   if ( m_codec )
+  {
+    m_codec->DeInit();
```

Comment:
I don't see what is here. Now you call Deinit before the destructor.
This change violates the principle of encapsulation. Why do you want to call DeInit from the outside?